dev/TreeView/ViewModel.cpp

```diff
@@ -310,15 +318,85 @@ void ViewModel::NodeCollapsed(const winrt::event_token token)
 
 void ViewModel::SelectAll()
 {
+    BeginSelectionChanges();
+    auto trackSelection = gsl::finally([this]() { EndSelectionChanges(); });
+
     UpdateSelection(m_originNode.get(), TreeNodeSelectionState::Selected);
 }
 
-void ViewModel::ModifySelectByIndex(int index, TreeNodeSelectionState const& state)
+void ViewModel::SelectItem(winrt::IInspectable const& item)    
+{
+    BeginSelectionChanges();
+    auto trackSelection = gsl::finally([this]() { EndSelectionChanges(); });
+
+    auto selectedItems = GetSelectedItems();
+    if (selectedItems.Size() > 0)
+    {
+        selectedItems.Clear();
+    }
```

Comment:
Do we somehow know we are in single selection mode here?

Reply:
No. I renamed the method however to be more explicit. This code was previously living in the `TreeView` type, I moved it here to be able to track selection changes. I think a refactoring of those methods (cf. your other comments) would help making this cleaner.